code/client/installhelper

```diff
@@ -0,0 +1,372 @@
+        cmd_out = subprocess.run(cmd, capture_output = True, check = False)
+        launchd_labels = split_launchctl_list(cmd_out)
+        # For each plist in /Library/LaunchAgents/ prefixed with LAUNCHD_PREFIX
+        for launch_agent in launch_agents:
+            try:
+                # Make sure content is valid, before proceeding to reload
+                agent_content = FoundationPlist.readPlist(launch_agent)
+            except FoundationPlist.NSPropertyListSerializationException:
+                break
+            # If the launch agent doesn't contain LimitLoadToSessionType or
+            # LimitLoadToSessionType exists and it contains Aqua
+            if (not agent_content.get('LimitLoadToSessionType') or 'Aqua' in
+                agent_content.get('LimitLoadToSessionType')):
+                # If the launch agent is to be loaded at the login window, get the Label
+                agent_label = agent_content.get('Label')
+                if agent_label:
+                    # Check to see if the launch agent is loaded, before stopping
+                    if agent_label in launchd_labels:
+                        # Stop the launch agent
+                        log("Stopping: %s" % launch_agent)
+                        cmd = ['/bin/launchctl', 'asuser', str(userid), '/bin/launchctl', 'bootout',
+                               'gui/' + str(userid) + '/' + agent_label]
+                        subprocess.call(cmd)
+                    # Load the launch agent
+                    log("Starting: %s" % launch_agent)
+                    cmd = ['/bin/launchctl', 'asuser', str(userid), '/bin/launchctl', 'bootstrap',
+                           'gui/' + str(userid), launch_agent]
+                    subprocess.call(cmd)
+
+
+def split_launchctl_list(cmd_out):
+    '''Takes launchctl list output and returns a list with just the labels'''
+
+    # Var declaration
+    launchd_labels = []
+
+    # Strip cmd_out to just the labels of the running launch agents
+    for launchd_item in cmd_out.stdout.decode().splitlines():
+        launchd_labels.append(launchd_item.split('\t')[2])
+
+    # Returns a list of launchd labels
+    return launchd_labels
+
+
+if __name__ == '__main__':
+
+    # Check to see if we're root
+    if os.geteuid() != 0:
+        print("You must run this as root!", file=sys.stderr)
+        sys.exit(1)
+
+    # If INSTALLHELPER_RUN_TYPE isn't set
+    if not os.environ.get('INSTALLHELPER_RUN_TYPE'):
+        # If we have no arguments passed or the arguments passed are not expected.. exit
+        if len(sys.argv) == 1 or not (sys.argv[1].lower() == 'appusage' or
+                                        sys.argv[1].lower() == 'launchd'):
+            print("Requires either \'appusage\' or \'launchd\' arguments to be passed.")
+            sys.exit(1)
+        # Set PASSED_ARG to the value passed
+        else:
+            PASSED_ARG = sys.argv[1].lower()
```

Comment:
If we really have to use a global, let's give it a more descriptive name throughout.

Reply:
This was changed to not using a global for this